source/docs/advanced/payments/hipay.md

```diff
@@ -0,0 +1,186 @@
+```
+3. Override the file that lets you register additional payments actions in Front-Commerce
+```
+mkdir -p my-module/web/theme/modules/Checkout/PlaceOrder
+cp -u node_modules/front-commerce/src/web/theme/modules/Checkout/PlaceOrder/getAdditionalActionComponent.js my-module/web/theme/modules/Checkout/PlaceOrder/getAdditionalActionComponent.js
+```
+4. Register the Adyen action
+```diff
+import None from "./AdditionalAction/None";
++import HiPayActions from "theme/modules/Checkout/PlaceOrder/AdditionalAction/HiPay/HiPayActions";
+
+const ComponentMap = {};
+
+const getAdditionalActionComponent = (paymentCode, paymentAdditionalData) => {
++  if (paymentCode.startsWith("hipay_")) {
++    return HiPayActions;
++  }
+  return ComponentMap?.[paymentCode] ?? None;
+};
+```
+5. register [custom Flash message components](/docs/advanced/features/flash-messages.html#Create-custom-flash-message-components) to display payment messages in an optimized way (**recommended**)
+```
+mkdir -p my-module/web/theme/modules/FlashMessages
+cp -u node_modules/front-commerce/src/web/theme/modules/FlashMessages/getFlashMessageComponent.js my-module/web/theme/modules/FlashMessages/getFlashMessageComponent.js
+```
+  and
+```diff
+import React from "react";
+import {
+  InfoAlert,
+  ErrorAlert,
+  SuccessAlert,
+} from "theme/components/molecules/Alert";
+import { BodyParagraph } from "theme/components/atoms/Typography/Body";
++import {
++  HiPayPaymentSuccess,
++  HiPayPaymentError,
++} from "theme/hipay/FlashMessage";
+
+// […]
+const ComponentMap = {
+  default: makeAlertMessageComponent(InfoAlert),
+  info: makeAlertMessageComponent(InfoAlert),
+  error: makeAlertMessageComponent(ErrorAlert),
+  success: makeAlertMessageComponent(SuccessAlert),
++  hipaySuccess: HiPayPaymentSuccess,
++  hipayError: HiPayPaymentError,
+};
+```
+6. register the HiPay payment methods labels
+```
+mkdir -p my-module/web/theme/modules/User/Order/OrderMethod/
+cp -u node_modules/front-commerce/src/web/theme/modules/User/Order/OrderMethod/PaymentMethodLabel.js my-module/web/theme/modules/User/Order/OrderMethod/PaymentMethodLabel.js
+```
+
+```diff
++ import HiPayMethodLabels from "theme/hipay/HiPayMethodLabels";
+
+const methodsMapping = {
+  ...
++  ...HiPayMethodLabels,
+};
+```
+
+### Update your CSPs
+
+To allow loading HiPay related remote resources:
+
+```diff
+// my-module/config/website.js
+  contentSecurityPolicy: {
+    directives: {
+-      scriptSrc: [],
++      scriptSrc: ["*.hipay.com"],
+-      frameSrc: [],
++      frameSrc: ["*.hipay.com"],
+-      styleSrc: [],
++      styleSrc: ["*.hipay.com"],
+      imgSrc: [],
+      fontSrc: [],
+-      connectSrc: [],
++      connectSrc: ["*.hipay.com"],
+      baseUri: [],
+    }
+  },
+```
+
+### That's it!
+
+<blockquote class="note">
+Please keep in mind that HiPay payment methods depends on the Customer country, the Cart amount and the Store currency. Different contexts could display different payment methods in the checkout.
```

Comment:
**question:** Is this correct in the current state?

Reply:
small mistake indeed, it can vary based on the customer country and currency, not based on the cart amount.